### PR TITLE
feat: Use IdentifierProvider for equality in radio-button-group

### PR DESCRIPTION
**Pull request by taefi**
Take `IdentifierProvider` into use use for determining the equality of the items in `vaadin-radio-button-flow`.
  
Fix: #273

### CODE REVIEW

**Comment by manolo** [comment]
I would change PR title to `Use IdentifierProvider for equality in radio-button-group`,  that is, starting with an imperative, otherwise the message in the  release notes would be weird.

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java [hunk 1]
```diff
@@ -519,6 +521,30 @@ private void validateSelectionEnabledState(PropertyChangeEvent event) {
...
+     * @return <code>true</code> if the instances are equal; otherwise
+     *         <code>false</code>
+     */
+    @Override
+    protected boolean valueEquals(T value1, T value2) {
+        if (value1 == null && value2 == null)
+            return true;
+        if (value1 == null || value2 == null)
+            return false;
+        return getItemId(value1).equals(getItemId(value2));
```
Comment (pleku) [inline vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java:545]: Don't you have the Vaadin Java coding conventions set up in the IDE ? Missing brackets ...

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupDataView.java [hunk 2]
```diff
@@ -63,4 +88,13 @@ public T getItem(int index) {
...
         return getItems().skip(index).findFirst().orElse(null);
     }
+
+    @Override
+    public void setIdentifierProvider(
+            IdentifierProvider<T> identifierProvider) {
+        super.setIdentifierProvider(identifierProvider);
+
+        if (identifierChangedCallback != null)
+            identifierChangedCallback.accept(identifierProvider);
```
Comment (pleku) [inline vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupDataView.java:98]: Missing brackets

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupListDataView.java [hunk 3]
```diff
@@ -44,4 +50,33 @@ public RadioButtonGroupListDataView(
...
+        this.identifierChangedCallback = identifierChangedCallback;
+    }
+
+    @Override
+    public void setIdentifierProvider(
+            IdentifierProvider<T> identifierProvider) {
+        super.setIdentifierProvider(identifierProvider);
+
+        if (identifierChangedCallback != null)
+            identifierChangedCallback.accept(identifierProvider);
```
Comment (pleku) [inline vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupListDataView.java:80]: missing brackets

**Review by pleku — CHANGES_REQUESTED** [review]

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java (hunk 1 above)
Comment (taefi) [reply vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java:545]: Actually, I do have that setup in my IDE (IntelliJ) but it is complaining about one-liner `if` statements to have the bracket. Although, personally I am a fan of having brackets always for one-line `if` and `for` statements, but I couldn't find it anywhere in the following places: 

- https://sites.google.com/a/vaadin.com/fw-flow-team-s-wiki/contributing-code/flow-way-of-devloping 
- https://sites.google.com/a/vaadin.com/fw-flow-team-s-wiki/contributing-code/code-style

Comment (taefi) [reply vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java:547]: Done.

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupDataView.java (hunk 2 above)
Comment (taefi) [reply vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupDataView.java:96]: Done.

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupListDataView.java (hunk 3 above)
Comment (taefi) [reply vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/dataview/RadioButtonGroupListDataView.java:78]: Done.

**Review by pleku — APPROVED** [review]

**Pull request merged**